src/json.cxx

```diff
@@ -15,3 +16,69 @@ std::string json_to_string(rapidjson::Document const &jd) {
   jd.Accept(w);
   return b1.GetString();
 }
+
+/// Merge the json structure v2 into a copy of v1.
+/// If the new key in v2 is not a object, it replaces the key in v1.
+/// Elements of arrays are also not merged, but the new array fully replaces
+/// the old one.
+rapidjson::Document merge(rapidjson::Value const &v1,
+                          rapidjson::Value const &v2) {
+  using namespace rapidjson;
+  using std::string;
+  using std::deque;
+  Document ret;
+  auto &a = ret.GetAllocator();
+  ret.CopyFrom(v1, a);
+  if (!v1.IsObject() || !v2.IsObject()) {
+    return ret;
+  }
+  struct to_copy_element {
+    string name;
+    Value const *v;
+  };
+  struct stack_element {
+    Value::ConstMemberIterator it1;
+    Value *v1;
+    Value const *v2;
+    deque<to_copy_element> to_copy;
+  };
+  deque<stack_element> stack;
+  stack.push_back(stack_element{ret.MemberBegin(), &ret, &v2});
+
+  while (stack.size() > 0) {
```

Comment:
The rapidjson api tends to be a bit on the verbose side, but I decided to add comments instead of taking out the loop.